onchain/rollups/contracts/consensus/IConsensus.sol

```diff
@@ -3,74 +3,68 @@
+/// These conditions can be checked prior to submitting a claim by listening to the events specified by this interface.
+/// @notice The acceptance criteria for claims may depend on the type of consensus, and is not transparent through this interface.
+/// For example, a claim may be accepted if it was...
+/// - submitted by an authority or;
+/// - submitted by the majority of a quorum or;
+/// - submitted and not proven wrong after some period of time.
 interface IConsensus {
-    /// @notice An application has joined the consensus' validation set.
-    /// @param application The application
-    /// @dev MUST be triggered on a successful call to `join`.
-    event ApplicationJoined(address application);
+    /// @notice A claim was submitted to the consensus.
+    /// @param submitter The submitter address
+    /// @param dapp The DApp contract address
+    /// @param inputRange The input range
+    /// @param epochHash The epoch hash
+    /// @dev The input range MUST NOT be empty.
+    /// @dev Overwrites any previous `ClaimSubmission(submitter, dapp, inputRange, _)`.
+    event ClaimSubmission(
+        address indexed submitter,
+        address indexed dapp,
+        InputRange inputRange,
+        bytes32 epochHash
+    );
+
+    /// @notice A claim was accepted by the consensus.
+    /// @param dapp The DApp contract address
+    /// @param inputRange The input range
+    /// @param epochHash The epoch hash
+    /// @dev The input range MUST NOT be empty.
+    /// @dev Overwrites any previous `ClaimAcceptance(dapp, inputRange, _)`.
+    /// @dev MUST be triggered whenever `getEpochHash(dapp, inputRange)` changes to `epochHash`.
+    /// @dev MUST be triggered after some `ClaimSubmission(_, dapp, inputRange, epochHash)`.
+    event ClaimAcceptance(
+        address indexed dapp,
+        InputRange inputRange,
+        bytes32 epochHash
+    );
 
-    /// @notice Get a specific claim regarding a specific DApp.
-    ///         The encoding of `_proofContext` might vary
-    ///         depending on the implementation.
-    /// @param _dapp The DApp address
-    /// @param _proofContext Data for retrieving the desired claim
-    /// @return epochHash_ The claimed epoch hash
-    /// @return firstInputIndex_ The index of the first input of the epoch in the input box
-    /// @return lastInputIndex_ The index of the last input of the epoch in the input box
-    function getClaim(
-        address _dapp,
-        bytes calldata _proofContext
-    )
-        external
-        view
-        returns (
-            bytes32 epochHash_,
-            uint256 firstInputIndex_,
-            uint256 lastInputIndex_
-        );
+    /// @notice Submit a claim to the consensus.
+    /// @param dapp The DApp contract address
+    /// @param inputRange The input range
+    /// @param epochHash The epoch hash
+    /// @dev MUST trigger a `ClaimSubmission` event.
```

Comment:
I'm following EIP-20 and EIP-1155, and saying when an event is triggered both in the event doc and in the function doc.

Reply:
I decided to remove it, following your suggestion.

Reply:
Yeah. It can be both too. I mentioned it because there were 2 events and only one of them had `/// @dev MUST be triggered`, and also 2 functions and only one of them had `/// @dev MUST trigger`